bundles/io/org.eclipse.smarthome.io.transport.mdns/src/main/java/org/eclipse/smarthome/io/transport/mdns/internal/MDNSClientImpl.java

```diff
@@ -24,25 +31,60 @@
 public class MDNSClientImpl implements MDNSClient {
     private final Logger logger = LoggerFactory.getLogger(MDNSClientImpl.class);
 
-    private JmDNS jmdns;
+    private Set<JmDNS> jmdnsInstances = new CopyOnWriteArraySet<>();
+
+    private static Set<InetAddress> getAllInetAddresses() {
+        final Set<InetAddress> addresses = new HashSet<>();
+        Enumeration<NetworkInterface> itInterfaces;
+        try {
+            itInterfaces = NetworkInterface.getNetworkInterfaces();
+        } catch (final SocketException e) {
+            return addresses;
+        }
+        while (itInterfaces.hasMoreElements()) {
+            final NetworkInterface iface = itInterfaces.nextElement();
+            try {
+                if (!iface.isUp() || iface.isLoopback()) {
+                    continue;
+                }
+            } catch (final SocketException ex) {
+                continue;
+            }
+            final Enumeration<InetAddress> itAddresses = iface.getInetAddresses();
+            while (itAddresses.hasMoreElements()) {
+                final InetAddress address = itAddresses.nextElement();
+                if (address.isLoopbackAddress() || address.isLinkLocalAddress()) {
+                    continue;
+                }
+                addresses.add(address);
+            }
+        }
+        return addresses;
+    }
 
     @Override
-    public JmDNS getClient() {
-        return jmdns;
+    public Set<JmDNS> getClientInstances() {
```

Comment:
I am not sure whether it is a valid approach or not, but it would look simpler for users. I don't think that we will have use cases where callers would need to have individual access to the different network interfaces. And if, we can maybe then add a dedicated additional method for it.